src/Storages/AlterCommands.cpp

```diff
@@ -524,6 +540,20 @@ void AlterCommand::apply(StorageInMemoryMetadata & metadata, const Context & con
                 settings_from_storage.push_back(change);
         }
     }
+    else if (type == RESET_SETTING)
+    {
+        auto & settings_from_storage = metadata.settings_changes->as<ASTSetQuery &>().changes;
+        for (const auto & setting_name : settings_resets)
+        {
+            auto finder = [&setting_name](const SettingChange & c) { return c.name == setting_name; };
+            auto it = std::find_if(settings_from_storage.begin(), settings_from_storage.end(), finder);
+
+            if (it != settings_from_storage.end())
+                settings_from_storage.erase(it);
+            else
+                throw Exception("Setting " + backQuote(setting_name) + " is not defined for a table", ErrorCodes::UNKNOWN_SETTING);
```

Comment:
current implementation implies the RESET SETTING is not idempotent.

Was thinking what is the best approach here ignore a missing setting or add `IF EXISTS` statement?

Reply:
Discussed in telegram. You should be able to always reset setting to default. Even if it was never adjusted for that table or was already reset before. 

Exception only for unknown / misspelled settings.

Reply:
the only problem with this approach, that there is no understanding of the table engine in AlterCommand. Need to pass a set of settings changes with additional information (setting has to be removed) to storage.

`RESET SETTING IF EXISTS` would be easier to implement.